src/TreeAnalyzer.php

```diff
@@ -52,8 +52,34 @@ public function __construct(PhpParser\Parser $parser, string $content, DocBlockF
         $this->collectDefinitionsAndReferences($this->sourceFileNode);
     }
 
+    private function collectDiagnostics($node)
+    {
+        if (($error = PhpParser\DiagnosticsProvider::checkDiagnostics($node)) !== null) {
+            $range = PhpParser\PositionUtilities::getRangeFromPosition($error->start, $error->length, $this->sourceFileNode->fileContents);
+
+            if ($error->kind == \Microsoft\PhpParser\DiagnosticKind::Error) {
+                $severity = DiagnosticSeverity::ERROR;
+            } else {
+                $severity = DiagnosticSeverity::WARNING;
+            }
```

Comment:
could you use a `switch` for this? that would it clearer that you are doing a 1:1 mapping

Reply:
Sure